custom_components/tuya_local/devices/kogan_lx10_vacuum.yaml

```diff
@@ -0,0 +1,332 @@
+          value: forward
+        - dps_val: backward
+          value: backward
+        - dps_val: left
+          value: left
+        - dps_val: right
+          value: right
+        - dps_val: stop
+          value: stop
+    - id: 112
+      name: locate # seek_robot
+      optional: true
+      type: boolean
+    - id: 114
+      name: volume
+      type: integer
+      range:
+        min: 0
+        max: 10
+    - id: 122
+      name: error # robot_fault
+      # TODO: correctly parse the bitfield
+      type: integer
+      # "values": "{\"label\":[\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"20\",\"21\",\"22\",\"23\",\"24\",\"25\",\"26\",\"27\",\"28\",\"29\",\"30\"],\"maxlen\":30}"
+    - id: 123
+      name: path_comm
+      optional: true
+      type: string
+    - id: 124
+      name: cmd_comm
+      optional: true
+      type: string
+    - id: 125
+      name: request_data
+      optional: true
+      type: string
+    - id: 126
+      name: comm_flag
+      optional: true
+      type: string
+    - id: 127
+      name: comm_raw
+      optional: true
+      type: string
+    - id: 128
+      name: message_report
+      type: string
+      mapping:
+        # TODO: figure out the meanings of these
+        - dps_val: '0'
+          value: '0'
+        - dps_val: '1'
+          value: '1'
+        - dps_val: '2'
+          value: '2'
+        - dps_val: '3'
+          value: '3'
+        - dps_val: '4'
+          value: '4'
+        - dps_val: '5'
+          value: '5'
+        - dps_val: '6'
+          value: '6'
+        - dps_val: '7'
+          value: '7'
+        - dps_val: '8'
+          value: '8'
+        - dps_val: '9'
+          value: '9'
+        - dps_val: '10'
+          value: '10'
+        - dps_val: '11'
+          value: '11'
+        - dps_val: '12'
+          value: '12'
+        - dps_val: '13'
+          value: '13'
+        - dps_val: '14'
+          value: '14'
+        - dps_val: '15'
+          value: '15'
+        - dps_val: '16'
+          value: '16'
+        - dps_val: '17'
+          value: '17'
+        - dps_val: '18'
+          value: '18'
+        - dps_val: '19'
+          value: '19'
+        - dps_val: '20'
+          value: '20'
+    - id: 130
+      name: sn
+      optional: true
+      type: string
+    - id: 131
+      name: uuid
+      optional: true
+      type: string
+    - id: 132
+      name: device_info
+      optional: true
+      type: string
+secondary_entities:
+  - entity: switch
+    name: charge_switch
+    category: config
+    icon: "mdi:power-plug"
+    dps:
+      - id: 103
+        name: charge_switch
+        type: boolean
+  - entity: number
+    name: cur_clean_time
+    category: diagnostic
+    icon: "mdi:clock-outline"
+    dps:
+      - id: 107
+        name: cur_clean_time
+        type: integer
+        unit: m
+  - entity: number
+    name: cur_clean_area
+    category: diagnostic
```

Comment:
category should be config for numbers, but if the entity is a sensor, then diagnostic is correct.